### PR TITLE
404 page added

### PR DESCRIPTION
**Pull request by Ankita297**
## Related Issue 

404 page added
Closes: #3 

### Describe the changes you've made

I have added the 404 page 
## Type of change

What sort of change have you made:

- [ x] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ x] Enhancement in the existing features
- [ x] Other

## Mention any unusual behaviour of your code (Write NA if not)
Any unusual behaviour of your code

## Checklist:

- [x ] My code follows the guidelines of this project.
- [ x] I have performed a self-review of my own code.
- [ x] I have commented my code, particularly whereever it was hard to understand.
- [ x] My changes generate no new warnings.
- [ x] I have verified/tested my code by running it locally.

## Additional Info (optional)
Any additional information you want to give
Gurujii com - Google Chrome 22-04-2021 19_47_41

Gurujii com - Google Chrome 22-04-2021 19_47_57

### CODE REVIEW

**Comment by Ankita297** [comment]
@sagar-barapatre @Namanl2001 Please review my PR .

**Comment by Ankita297** [comment]
Sir i  have added the Navbar separtely on both pages because homepage needed parallax effect and 404 page does not need it .
But as we use Navbar on app.js it will be render on both page .

**Comment by sagar-barapatre** [comment]
Okay this looks fine to me.

**Review by sagar-barapatre — APPROVED** [review]
Just check once again that "You went too far buddy!" looks more formal or not. Rest all is good.

**Comment by Ankita297** [comment]
@sagar-barapatre According to me its okay but if you want we can change it .

**Comment by Ankita297** [comment]
@sagar-barapatre please tell what change do you want ?

**Comment by sagar-barapatre** [comment]
I was thinking of something like "Oops, the page you're looking for doesn't exists".

**Comment by Ankita297** [comment]
Home_external tutor by akshay1027 · Pull Request #122 · Namanl2001_MERN-Gurujii-dev - Google Chrome 23-04-2021 19_00_29

@Namanl2001 @sagar-barapatre is it okay now ?

File: src/components/home.jsx [hunk 1]
```diff
@@ -6,6 +6,9 @@ import Filter from './filter';
 import { Container, Row } from 'bootstrap-4-react';
 import { Progress } from 'bootstrap-4-react';
 import ScrollToTop from './scrollToTop';
+import Board from './board.jpg';
```
Comment (Namanl2001) [inline src/components/home.jsx:9]: could you please move all the images in `/src/component` to `/src/assets`? And update the paths in the files
ps: you have to make a new directory `assets`

**Review by Namanl2001 — CHANGES_REQUESTED** [review]
Also, verify the 404 page for a mobile device after you have updated the err msg. Thanks

File: src/components/home.jsx (hunk 1 above)
Comment (Ankita297) [reply src/components/home.jsx]: done @Namanl2001 .

**Comment by Ankita297** [comment]
Gurujii com - Google Chrome 24-04-2021 10_26_14

**Comment by Ankita297** [comment]
@Namanl2001 @sagar-barapatre Please review it now .

**Comment by Ankita297** [comment]
@Namanl2001  @sagar-barapatre  any other changes required ? The mobile view is fine .

**Review by sagar-barapatre — APPROVED** [review]
Yeah! This looks fine to me

**Comment by Ankita297** [comment]
@Namanl2001  is it okay now ?

**Review by Namanl2001 — CHANGES_REQUESTED** [review]
- The alert (green one from above on navbar) coming on the error page should not come.
- The Home option on navbar should take us to home page, when we click on it from error page.
- This section in screenshot should be as it is now on the master branch. Please don't change it
 
image

**Comment by Ankita297** [comment]
https://user-images.githubusercontent.com/67833077/116275682-79cd4100-a7a1-11eb-97a8-1c8339db4682.mp4

@sagar-barapatre @Namanl2001 is it okay now ?

**Comment by Namanl2001** [comment]
In the video attached I'm able to see 1st and 2nd point implemented but what about 3rd point mentioned above?

**Comment by Ankita297** [comment]
@Namanl2001 Sir i have not changed the part related to filter .

**Comment by Ankita297** [comment]
Thus the part of filter is shown when data  intialized is complete  ?
Gurujii com - Google Chrome 27-04-2021 22_06_38

**Comment by Namanl2001** [comment]
I guess you haven't setup the mongodb for this project and you are right the part of filter is shown when data intialized is complete

**Comment by Ankita297** [comment]
Yes i have not set up mongodb part do i have to do it . Actually i was not able to set it up  and it was working fine so i thought it was not necessary .

**Comment by Namanl2001** [comment]
Yes it's needed if you want to test the frontend part. It's simple you can google for it. Else you can also put your query on discord channel.

**Comment by Ankita297** [comment]
https://user-images.githubusercontent.com/67833077/116291187-ef410d80-a7b1-11eb-8cf8-e4d7118a75c5.mp4

**Comment by Ankita297** [comment]
@Namanl2001 Please review it now .

File: src/components/E404.jsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+const E404 = () => {
+  return (
+    <div>
+      <Navbar />
+      <Chatbot />
+      <h1 className='error_heading'>
+        Oops &#44; the page you&apos;re looking for doesn&apos;t exist
+      </h1>
+      <img src={Img404} className='error_img'></img>
```
Comment (Namanl2001) [inline src/components/E404.jsx:15]: add an `alt` tag.

File: src/components/home.jsx [hunk 3]
```diff
@@ -45,6 +49,20 @@ class home extends Component {
...
+          <br />
+          <br />
+          <div className='center'>
+            <img src={Board} alt='' />
+          </div>
+          <br />
+          <br />
+          <br />
+        </div>
+        <Chatbot />
```
Comment (Namanl2001) [inline src/components/home.jsx:65]: Add one `<br />` at line 76 of this file.
and one at line 68 of `filter.jsx`

File: src/components/alert.jsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      <div>
+        {this.state.hidden ? (
+          ' '
+        ) : (
+          <div
+            className='alert alert-success alert-dismissible fade show myalert'
+            // style="margin-bottom:0"
+          >
+            <strong>Great Landing !</strong> Teachers have to signin to make
+            profile.
```
Comment (Namanl2001) [inline src/components/alert.jsx:23]: enclose this in `<p>...</p>` Else time is also being shown in alert.

**Review by Namanl2001 — CHANGES_REQUESTED** [review]
Small nitpicks

**Comment by Ankita297** [comment]
@Namanl2001 @sagar-barapatre  changes done Sir .

**Review by Namanl2001 — APPROVED** [review]
LGTM! Thanks

**Pull request merged**